src/mscorlib/shared/System/IO/FileStream.cs

```diff
@@ -81,8 +81,31 @@ public FileStream(IntPtr handle, FileAccess access, bool ownsHandle, int bufferS
 
         [Obsolete("This constructor has been deprecated.  Please use new FileStream(SafeFileHandle handle, FileAccess access, int bufferSize, bool isAsync) instead, and optionally make a new SafeFileHandle with ownsHandle=false if needed.  http://go.microsoft.com/fwlink/?linkid=14202")]
         public FileStream(IntPtr handle, FileAccess access, bool ownsHandle, int bufferSize, bool isAsync)
-            : this(new SafeFileHandle(handle, ownsHandle), access, bufferSize, isAsync)
         {
+            SafeFileHandle safeHandle = new SafeFileHandle(handle, ownsHandle: ownsHandle);
+            try
+            {
+                ValidateAndInitFromHandle(safeHandle, access, bufferSize, isAsync);
+            }
+            catch
+            {
+                // We don't want to take ownership of closing passed in handles
+                // *unless* the constructor completes successfully.
+                GC.SuppressFinalize(safeHandle);
+
+                // This would also prevent Close from being called, but is unnecessary
+                // as we've removed the object from the finalizer queue.
+                //
+                // safeHandle.SetHandleAsInvalid();
```

Comment:
> The calling code is handing off a raw IntPtr to the FileStream saying "it's now your responsibility to close this, nothing else will".

My argument on this one is that people don't expect objects that aren't fully constructed to have ownership of closing passed in objects, period.

> That's very different from the ctors that take a SafeHandle, where the SafeHandle itself knows whether it owns the IntPtr, and where worst case it'll have its finalizer handle the cleanup.

I agree that there is a safety net there that isn't in place with the IntPtr, but it feels off to me to not be consistent. That said, we have depreciated the IntPtr constructors... @jkotas, @terrajobst, @weshaggard: Do you have any feedback on the right pattern here? There are likely other places where we have destructors that need hardened against constructor exceptions- I'd like to vet the approach.